### DISCUSSION TITLE
InsecureRequestWarning when committing via wlc

### DISCUSSION THREAD

**Discussion by Alireza-Farahani** [WeblateOrg/weblate#6132] [Q&A] [answered]
### Context
Our hosted Weblate is behind a Nginx basic auth and our projects are public and basic auth is excluded from `/git/` paths. I can clone Weblate repo using `git clone https://weblate.example.com/git/proj/component.git`
Our weblate version is `4.5.1`

### Question
I installed wlc and created config file in `.config/weblate`

```
[weblate]
url = https://user:pass@weblate.example.com/api/

[keys]
https://user:pass@weblate.example.com/api/ = <API key>
```
I can successfully run `wlc ls` but when I try `wlc commit <proj>`, it gives me:
```
/usr/lib/python3/dist-packages/urllib3/connectionpool.py:999: InsecureRequestWarning: Unverified HTTPS request is being made to host 'weblate.example.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
API key configured for https://user:pass@weblate.example.com/api/ was rejected by server.
```
What could be the problem?

**Comment by nijel** [comment, marked as answer]
Passing user/pass definitely probably won't work - the key is also passed as an Authorization header, so there is no way to include additional credentials in the HTTP request. Exclude the basic auth for the /api/ as well.

The warning might indicate that different server URL is used in that case (the default 127.0.0.1), try running wlc with --debug to see that.

**Reply by Alireza-Farahani** [reply]
We ended up disabling basic auth and instead chose to disable registration and made projects private.

Thank you nijel
